### PR TITLE
[mono] Added null-checks for LDFLDA on pointers.

### PR DESCRIPTION
**Pull request by jandupej**
Addresses the issue #77647 by adding a null check when translating the CIL opcode LDFLDA operating on a pointer into Mono IR. The appropriate unit test is reenabled for Mono.

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -6034,19 +6034,10 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 					/* TODO: metadata-update: implement me. If it's an added field, emit a call to the helper method instead of MINT_LDFLDA_UNSAFE */
 					g_assert (!m_field_is_from_update (field));
 					int foffset = m_class_is_valuetype (klass) ? m_field_get_offset (field) - MONO_ABI_SIZEOF (MonoObject) : m_field_get_offset (field);
-					if (td->sp->type == STACK_TYPE_O) {
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/transform.c:6037]: I don't think `STACK_TYPE_MP` should be null. Wouldn't the following more conservative fix work instead:
```
if (td->sp->type == STACK_TYPE_O || td->sp->type == STACK_TYPE_I) {
```

Comment (jandupej) [reply src/mono/mono/mini/interp/transform.c:6037]: I need it to perform a null-check with `STACK_TYPE_MP` as LDFLDA must throw if it is given a null pointer. I don't think this change would achieve that.

Comment (BrzVlad) [reply src/mono/mono/mini/interp/transform.c:6037]: I checked locally the testcase from `src/tests/JIT/Regression/JitBlue/Runtime_77640/Runtime_77640.cs` and the type of the pointer is indeed `STACK_TYPE_I` not `STACK_TYPE_MP`. Did you run also some other test ?

Comment (jandupej) [reply src/mono/mono/mini/interp/transform.c:6037]: Your fix seem to do the trick, as well. Strange, I thought that pointers are `STACK_TYPE_MP`.

**Review by BrzVlad — APPROVED** [review]

File: src/mono/mono/mini/interp/transform.c (hunk 1 above)
Comment (SingleAccretion) [reply src/mono/mono/mini/interp/transform.c:6037]: Managed pointers can also be null.

Example:
```cs
Problem(ref *(StructWithField*)null);

static ref int Problem(ref StructWithField p) => ref p.Field;

struct StructWithField
{
    public int Field;
}
```
(A more real-world example would involved `default` ref structs with `ref` fields)

**Pull request merged**